assert/assertions.go

```diff
@@ -1928,6 +1928,23 @@ func typeAndKind(v interface{}) (reflect.Type, reflect.Kind) {
 	return t, k
 }
 
+func isStructWithUnexportedMapField(t reflect.Type) bool {
+	if t.Kind() != reflect.Struct {
+		return false
+	}
+
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		if !field.IsExported() &&
+			field.Type.Kind() == reflect.Map &&
+			field.Type.Key().Kind() == reflect.Array {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
Based on the code the discussion that happened here


And based on the code of go-spew

https://github.com/stretchr/testify/pull/1816#issuecomment-3493594441

Reply:
I feel like started writting something, but I posted before putting all together.

So here, I was spotting the code here limited to unexported field that are maps with 2 elements, but go-spew code would panic when dealing with Interface and here, maybe a better fix would be to use CanInterface and check nothing else.

I'm unsure what other code examples could panic, but maybe it could be considered as a better approach than the "recover" I suggested.